pkg/deploy/deployment.go

```diff
@@ -109,16 +109,16 @@ func OverrideDeployment(
 		var overrideContainerSettings *chev2.Container
 		container := &deployment.Spec.Template.Spec.Containers[index]
 
-		if overrideDeploymentSettings != nil {
-			if len(deployment.Spec.Template.Spec.Containers) != 1 {
+		if overrideDeploymentSettings != nil && len(overrideDeploymentSettings.Containers) > 0 {
+			if len(deployment.Spec.Template.Spec.Containers) == 1 {
+				overrideContainerSettings = &overrideDeploymentSettings.Containers[0]
+			} else {
```

Comment:
@tolusha why do we explicitly handle the case with one container in deployment? can't it be part of the for loop below?

Reply:
Basically for simplifying override settings. 
For instance, it is enough to override container without specifying its name in case of che-server, registries or dashboard because it can be the only one container in the pod:
```yaml
spec:
  components:
    <COMPONENT>:
      deployment:
        containers:
          - image: <COMPONENT_IMAGE>
```

But in case of gateway, we have to specify containers' names, because there can be 4 containers in the pod and we need a way to figure out which override settings belongs to which container.
```yaml
spec:
  networking:
    auth:
      gateway:
        deployment:
          containers:
            - name: gateway
              ...
            - name: configbump
              ...
            - name: oauth-proxy
              ...
            - name: kube-rbac-proxy
              ...
```

Reply:
makes sense